### PR TITLE
Migrate RCEntitlementInfos

### LINKED ISSUE #554: Migrate EntitlementInfos
Blocked on #553

### PR DESCRIPTION
**Pull request by taquitos**
Resolves #554

### CODE REVIEW

File: Purchases/Public/RCEntitlementInfos.h [hunk 1]
```diff
@@ -1,36 +0,0 @@
...
- Dictionary of all EntitlementInfo (`RCEntitlementInfo`) objects (active and inactive) keyed by entitlement identifier. This dictionary can also be accessed by using an index subscript on EntitlementInfos, e.g. `entitlementInfos[@"pro_entitlement_id"]`.
- */
-@property (readonly) NSDictionary<NSString *, RCEntitlementInfo *> *all;
-
-/**
- Dictionary of active EntitlementInfo (`RCEntitlementInfo`) objects keyed by entitlement identifier.
- */
-@property (readonly) NSDictionary<NSString *, RCEntitlementInfo *> *active;
-
-/// :nodoc:
```
Comment (aboedo) [inline Purchases/Public/RCEntitlementInfos.h:31]: [unrelated to this PR] I hadn't noticed this before. We should have had a doc for this one. I guess its purpose is obvious if you're used to iOS, but probably not for anyone else

File: PurchasesCoreSwift/Public/EntitlementInfos.swift [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            guard let entitlement = entitlement as? [String: Any] else {
+                return
+            }
+
+            guard let productIdentifier = entitlement["product_identifier"] as? String else {
+                return
+            }
+
+            let productData = purchasesData[productIdentifier]
+            guard let productData = productData as? [String: Any] else {
```
Comment (aboedo) [inline PurchasesCoreSwift/Public/EntitlementInfos.swift:49]: perhaps these could be grouped into a single clause? I don't have a strong preference, though, readability for this is fine (and miles better than what it used to be before the migration)

File: PurchasesCoreSwift/Public/EntitlementInfos.swift [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                return
+            }
+            entitlementInfos[identifier] = EntitlementInfo(entitlementId: identifier,
+                                                                  entitlementData: entitlement,
+                                                                  productData: productData,
+                                                                  dateFormatter: dateFormatter,
+                                                                  requestDate: requestDate)
```
Comment (aboedo) [inline PurchasesCoreSwift/Public/EntitlementInfos.swift:56]: indentation here is a bit off

File: PurchasesCoreSwift/Public/EntitlementInfos.swift (hunk 2 above)
Comment (taquitos) [reply PurchasesCoreSwift/Public/EntitlementInfos.swift:49]: To me, this reads better than grouping them all.

File: PurchasesCoreSwift/Public/EntitlementInfos.swift (hunk 3 above)
Comment (taquitos) [reply PurchasesCoreSwift/Public/EntitlementInfos.swift:56]: 🤦‍♂️

**Pull request merged**